src/components/CartDrawer.tsx

```diff
@@ -0,0 +1,67 @@
+import { Link } from "@chakra-ui/next-js";
+import {
+  Box,
+  Drawer,
+  DrawerBody,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerOverlay,
+  Flex,
+  HStack,
+  Image,
+  Text,
+} from "@chakra-ui/react";
+
+import { ActionButton, OutlineButton } from "~/components/Button";
+import { useCart } from "~/contexts/cart";
+
+// TODO: add quantity selector, and price calculation
+type CartDrawerProps = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+export const CartDrawer = ({ isOpen, onClose }: CartDrawerProps) => {
+  const { removeFromCart, cartItems } = useCart();
+
+  return (
+    <Drawer isOpen={isOpen} placement="right" size={"md"} onClose={onClose}>
+      <DrawerOverlay />
+      <DrawerContent>
+        <DrawerCloseButton />
+        <DrawerHeader>Shopping Cart</DrawerHeader>
+
+        <DrawerBody>
+          {cartItems.length === 0 && <Text>It is empty in here</Text>}
```

Comment:
suggestion: maybe `Cart` for title and `Your cart is empty.` centered + a bit bigger